runtime/block_reduction.cu

```diff
@@ -63,14 +113,17 @@ __device__ void blockReduce(
   int np2 = 1 << (31 - __clz(reduction_size));
 
   if (reduction_tid < np2 && reduction_tid + np2 < reduction_size) {
-    reduction_op(shared_mem[smem_offset], shared_mem[smem_offset + np2]);
+    reduction_op(
+        shared_mem[smem_offset], shared_mem[smem_offset + np2 * peer_stride]);
```

Comment:
Is the new layout always a net-gain?
`peer_stride != 0` here would mess with the shared load in the tree reduction `shared_mem[smem_offset + np2 * peer_stride]`. With outer reduction, `peer_stride == blockDim.x`, I'm surprised it removed all the bank conflict....

Reply:
These two smem load are 2 sepearte transactions. added a figure for reference:
![image](https://github.com/NVIDIA/Fuser/assets/116412316/a7687c88-874d-4c08-bc58-141516de7a56)